Fix small bugs/edge cases in the parthenon frontend

Yurlungur opened the pull request:
<!--Thank you so much for your PR! To help us review, fill out the form
to the best of your ability.  Please make use of the development guide at
http://yt-project.org/docs/dev/developing/index.html-->

<!--Provide a general summary of your changes in the title above, for
example "Raises ValueError on Non-Numeric Input to set_xlim".  Please avoid
non-descriptive titles such as "Addresses issue #8576".-->

<!--If you are able to do so, please do not create the PR out of main, but out
of a separate branch. -->

## PR Summary

We recently *finally* refactored curvilinear coordinates in Parthenon and the ordering for cylindrical coordinates was normalized to the standard choice in yt. This MR fixes that in parthenon frontend. I also noticed that the yt frontend was not properly handling variables of tensor rank greater than 1 (i.e., 2d matrices on a mesh). This MR adds a one-line fix that gets that working.

<!--Please provide at least 1-2 sentences describing the pull request in
detail.  Why is this change required?  What problem does it solve?-->

<!--If it fixes an open issue, please link to the issue here.-->

## PR Checklist

<!-- Note that some of these check boxes may not apply to all pull requests -->
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

<!--We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or the
recommended next step seems overly demanding , or if you would like help in
addressing a reviewer's comments.  And please ping us if you've been waiting
too long to hear back on your PR.-->

Yurlungur commented on yt/frontends/parthenon/io.py:
```diff
@@ -52,7 +52,10 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
                     start = gs[0].id - gs[0]._id_offset
                     end = gs[-1].id - gs[-1]._id_offset + 1
-                    data = ds[start:end, fdi, :, :, :].transpose()
+                    if len(ds.shape) == 4:
+                        data = ds[start:end, :, :, :].transpose()
+                    else:
+                        data = ds[start:end, fdi, :, :, :].transpose()
```
Parthenon's python reader can always split a variable into a flat index space, so I *think* this is now generic.

chrishavlin commented on yt/frontends/parthenon/data_structures.py:
```diff
@@ -164,10 +164,7 @@ def __init__(
-            axis_order = ("r", "theta", "z")
-        else:
-            axis_order = None
+        axis_order = None
```
this would break things for the old format, ya? anyway to detect the new format so you could make this change backward compatible?

Yurlungur replied:
Thanks for the review! See @pgrete 's comment below. The old version was for a draft of curvilinear coordinates that never made it into the code, so this isn't actually breaking. I think it's safe to merge this without a version check. (That said parthenon does have version checking.)

pgrete commented:
@Yurlungur do you have any data to share with those coordinate that could be added to the regression tests?
I think some choice were specific to the (not merged) changes in AthenaPK, so I'd be happy to follow the new "standard" way if we merge non-Cart coords in AthenaPK.

pgrete commented:
btw if you quickly want to run local tests, just 
```
nosetests -v --with-answer-testing --local --local-dir $HOME/src/yt-test-data --answer-store --answer-name=local-parthenon yt.frontends.parthenon
```
where `$HOME/src/yt-test-data` (or whatever folder of you choice) contains the data in the zip file (including the folder) -- and eventually data from https://yt-project.org/data/

Yurlungur commented:
> Dataset PR is here: [yt-project/website#141](https://github.com/yt-project/website/pull/141)
> 
> Also, I added a skeleton on how to test the data (currently just checking for the field name). Any creative suggestion on how to test the data itself @Yurlungur ?

Sorry for the delay. Been traveling. I added a check that the cell width in theta is 2pi. (This was being incorrectly overwritten, by the way, which I fixed.) I also added a check for the total mass.

pgrete commented:
@chrishavlin this now looks good to me.
Who can upload the new reference data to the yt website and accept https://github.com/yt-project/website/pull/141 so that the tests can pass?

chrishavlin commented:
@pgrete I think we need @Xarthisius to move the new dataset from https://github.com/yt-project/website/pull/141 to where it needs to go... but the test server that runs the full test suite is currently on the fritz, so the full nose test suite will not run at present (right now, the only tests that are running are those that do not require sample datasets). Doesn't hurt to have the dataset in place for when we revamp the test server, but not sure when that will be. In the meantime, I'll  double check that the new tests run for me locally right now and we should be good to merge.

chrishavlin commented on yt/frontends/parthenon/data_structures.py:
```diff
@@ -164,10 +160,7 @@ def __init__(
-            axis_order = ("r", "theta", "z")
-        else:
-            axis_order = None
+        axis_order = None
```
You can simply remove this entirely (and remove the corresponding line in the Dataset.__init__() call, line 172 below) and rely on the built in default handling. 

```suggestion
```

chrishavlin reviewed (commented):
I did run the existing answer tests locally and confirmed that they passed -- was not able to run the new test due to an error in unpacking the new test file (see https://github.com/yt-project/website/pull/141#issuecomment-3377551372 ). 

One final minor code suggestion if you want to fix it, but IMO this is good to merge. I'll hit approve once we get the new test file figured out.

Yurlungur commented:
🎊 Thanks for your feedback and helping pull it all through @chrishavlin @pgrete !
